lib/plek.rb

```diff
@@ -37,8 +34,14 @@ class NoConfigurationError < StandardError; end
   #   +GOVUK_APP_DOMAIN_EXTERNAL+ and if that is unavailable the parent domain
   #   will be used
   def initialize(domain_to_use = nil, external_domain = nil)
-    self.parent_domain = domain_to_use || env_var_or_dev_fallback("GOVUK_APP_DOMAIN", DEV_DOMAIN)
-    self.external_domain = external_domain || ENV["GOVUK_APP_DOMAIN_EXTERNAL"] || parent_domain
+    truth_re = /^[1tTyY]/
+    @parent_domain = domain_to_use || env_var_or_dev_fallback("GOVUK_APP_DOMAIN", DEV_DOMAIN)
+    @external_domain = external_domain || ENV.fetch("GOVUK_APP_DOMAIN_EXTERNAL", @parent_domain)
+    @host_prefix = ENV.fetch("PLEK_HOSTNAME_PREFIX", "")
+    # TODO: switch to a data structure with with constant-time lookups if we
+    # end up with more than a tiny number of unprefixable_hosts entries.
```

Comment:
I don't think this is needed as a todo as it seems unlikely, it feels more of an implementation note you may want to leave in the commit message?